libcudacxx/include/cuda/std/__cmath/min_max.h

```diff
@@ -0,0 +1,227 @@
+
+#ifndef _LIBCUDACXX___CMATH_MIN_MAX_H
+#define _LIBCUDACXX___CMATH_MIN_MAX_H
+
+#include <cuda/std/detail/__config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#include <cuda/std/__cmath/common.h>
+#include <cuda/std/__type_traits/enable_if.h>
+#include <cuda/std/__type_traits/is_arithmetic.h>
+#include <cuda/std/__type_traits/promote.h>
+
+#include <nv/target>
+
+_LIBCUDACXX_BEGIN_NAMESPACE_STD
+
+// fmax
+
+_CCCL_NODISCARD _LIBCUDACXX_HIDE_FROM_ABI float fmax(float __x, float __y) noexcept
+{
+#if defined(_CCCL_BUILTIN_FMAX)
+  return _CCCL_BUILTIN_FMAXF(__x, __y);
+#else // ^^^ _CCCL_BUILTIN_FMAX ^^^ / vvv !_CCCL_BUILTIN_FMAX vvv
+  return ::fmaxf(__x, __y);
+#endif // !_CCCL_BUILTIN_FMAX
+}
+
+_CCCL_NODISCARD _LIBCUDACXX_HIDE_FROM_ABI float fmaxf(float __x, float __y) noexcept
+{
+#if defined(_CCCL_BUILTIN_FMAX)
+  return _CCCL_BUILTIN_FMAXF(__x, __y);
+#else // ^^^ _CCCL_BUILTIN_FMAX ^^^ / vvv !_CCCL_BUILTIN_FMAX vvv
+  return ::fmaxf(__x, __y);
+#endif // !_CCCL_BUILTIN_FMAX
+}
+
+_CCCL_NODISCARD _LIBCUDACXX_HIDE_FROM_ABI double fmax(double __x, double __y) noexcept
+{
+#if defined(_CCCL_BUILTIN_FMAX)
+  return _CCCL_BUILTIN_FMAX(__x, __y);
+#else // ^^^ _CCCL_BUILTIN_FMAX ^^^ / vvv !_CCCL_BUILTIN_FMAX vvv
+  return ::fmax(__x, __y);
+#endif // !_CCCL_BUILTIN_FMAX
+}
+
+#if !defined(_LIBCUDACXX_HAS_NO_LONG_DOUBLE)
+_CCCL_NODISCARD _LIBCUDACXX_HIDE_FROM_ABI long double fmax(long double __x, long double __y) noexcept
+{
+#  if defined(_CCCL_BUILTIN_FMAX)
+  return _CCCL_BUILTIN_FMAXL(__x, __y);
+#  else // ^^^ _CCCL_BUILTIN_FMAX ^^^ / vvv !_CCCL_BUILTIN_FMAX vvv
+  return ::fmaxl(__x, __y);
+#  endif // !_CCCL_BUILTIN_FMAX
+}
+_CCCL_NODISCARD _LIBCUDACXX_HIDE_FROM_ABI long double fmaxl(long double __x, long double __y) noexcept
+{
+#  if defined(_CCCL_BUILTIN_FMAX)
+  return _CCCL_BUILTIN_FMAXL(__x, __y);
+#  else // ^^^ _CCCL_BUILTIN_FMAX ^^^ / vvv !_CCCL_BUILTIN_FMAX vvv
+  return ::fmaxl(__x, __y);
+#  endif // !_CCCL_BUILTIN_FMAX
+}
+#endif // !_LIBCUDACXX_HAS_NO_LONG_DOUBLE
+
+#if defined(_LIBCUDACXX_HAS_NVFP16)
+_CCCL_NODISCARD _LIBCUDACXX_HIDE_FROM_ABI __half fmax(__half __x, __half __y) noexcept
+{
+  NV_IF_ELSE_TARGET(NV_IS_DEVICE,
+                    (return ::__hmax(__x, __y);),
+                    (return __float2half(_CUDA_VSTD::fmaxf(__half2float(__x), __half2float(__y)));))
+}
+template <class _A1, enable_if_t<_CCCL_TRAIT(is_arithmetic, _A1), int> = 0>
+_CCCL_NODISCARD _LIBCUDACXX_HIDE_FROM_ABI __promote_t<float, _A1> fmax(__half __x, _A1 __y) noexcept
+{
+  return _CUDA_VSTD::fmaxf(__half2float(__x), __y);
+}
+
+template <class _A1, enable_if_t<_CCCL_TRAIT(is_arithmetic, _A1), int> = 0>
+_CCCL_NODISCARD _LIBCUDACXX_HIDE_FROM_ABI __promote_t<_A1, float> fmax(_A1 __x, __half __y) noexcept
```

Comment:
We need them because neither `__half` nor `__nv_bflaot16` satisfy `is_arithmetic`

I was pondering introducing a new trait that includes those but I am not sure I want to do that here and this seemed like the easier solution